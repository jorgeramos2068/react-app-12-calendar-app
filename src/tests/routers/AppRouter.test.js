import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { MemoryRouter} from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import AppRouter from '../../routers/AppRouter';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Tests in AppRouter router', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('Should show loading message', () => {
    const initialState = {
      auth: {
        checking: true,
      }
    };
    const store = mockStore(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <AppRouter />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h5').exists()).toBe(true);
  });

  test('Should show public route when the user is not logged', () => {
    const initialState = {
      auth: {
        checking: false,
        uid: null
      }
    };
    const store = mockStore(initialState);
    const wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <AppRouter />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.login-container').exists()).toBe(true);
  });

  test('Should show private route when the user is logged', () => {
    const initialState = {
      auth: {
        checking: false,
        uid: 'test_uid',
        name: 'test_name'
      },
      calendar: {
        events: [],
        activeEvent: null
      },
      ui: {
        modalOpen: false
      }
    };
    const store = mockStore(initialState);
    store.dispatch = jest.fn();
    const wrapper = mount(
      <div id="root">
        <Provider store={store}>
          <MemoryRouter>
            <AppRouter />
          </MemoryRouter>
        </Provider>
      </div>
    );
    expect(wrapper.find('.calendar-screen').exists()).toBe(true);
  });
});
