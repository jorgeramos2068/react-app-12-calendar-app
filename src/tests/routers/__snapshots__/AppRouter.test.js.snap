// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in AppRouter router Should show loading message 1`] = `
<h5>
  Loading...
</h5>
`;

exports[`Tests in AppRouter router Should show public route when the user is not logged 1`] = `
<div>
  <div
    className="container login-container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6 login-form-1"
      >
        <h3>
          Login
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="loginEmail"
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="loginPassword"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        className="col-md-6 login-form-2"
      >
        <h3>
          Register
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="registerName"
              onChange={[Function]}
              placeholder="Name"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="registerEmail"
              onChange={[Function]}
              placeholder="Email"
              type="email"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="registerPassword"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="registerPassword2"
              onChange={[Function]}
              placeholder="Retype your password"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Create account"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
